jdaviz/app.py

```diff
@@ -68,18 +68,30 @@
 class UnitConverterWithSpectral:
 
     def equivalent_units(self, data, cid, units):
+        # to debug
+        # print("eqv unit cid : ", cid, ", units : ", units)
+
         if cid.label == "flux":
             eqv = u.spectral_density(1 * u.m)  # Value does not matter here.
             list_of_units = set(list(map(str, u.Unit(units).find_equivalent_units(
-                include_prefix_units=True, equivalencies=eqv))) + [
-                    'Jy', 'mJy', 'uJy',
+                include_prefix_units=True, equivalencies=eqv)))
+                + [
+                    'Jy', 'mJy', 'uJy', 'MJy',
                     'W / (m2 Hz)', 'W / (Hz m2)',  # Order is different in astropy v5.3
                     'eV / (s m2 Hz)', 'eV / (Hz s m2)',
                     'erg / (s cm2)',
                     'erg / (s cm2 Angstrom)', 'erg / (Angstrom s cm2)',
                     'erg / (s cm2 Hz)', 'erg / (Hz s cm2)',
                     'ph / (s cm2 Angstrom)', 'ph / (Angstrom s cm2)',
                     'ph / (s cm2 Hz)', 'ph / (Hz s cm2)'
+                ]
+                + [
+                    'Jy / sr', 'mJy / sr', 'uJy / sr', 'MJy / sr',
+                    'W / (m2 Hz sr)',
+                    'eV / (s m2 Hz sr)',
+                    'erg / (s cm2 sr)',
+                    'erg / (s cm2 Angstrom sr)', 'erg / (s cm2 Hz sr)',
+                    'ph / (s cm2 Angstrom sr)', 'ph / (s cm2 Hz sr)'
```

Comment:
@astrofrog I do think this update to the equivalent units set needed to be addressed, however the tracebacks regarding `IncompatibleAttribute` and `UnitConversionError` aren't resolved from this addition. 

Both tracebacks last jdaviz trigger for this diff is in unit_conversions.py at the two statements: `self.spectrum_viewer.state.y_display_unit = str(spec_units)`. Both occur whether or not I use astropy's custom equivalencies. Everything updates in the viewer/display_units, but the tracebacks proceed. 

Also, it still may be needed to determine, but with this change,  y-limits on the profile viewer are not updating with this change (but this may change if the tracebacks are resolved).